### ISSUE TITLE
Bug initializing vfx static variables 

### ISSUE THREAD

**Issue by s-perron** [GPUOpen-Drivers/llpc#769]
I believe there is a problem initializing static variable because one depends on the other, but there is no guarantee of which constructor will be run first.

The varaible [`renderSectionParserInit`](https://github.com/GPUOpen-Drivers/llpc/blob/dev/tool/vfx/vfxRenderSection.cpp#L73) is a static.  It calls the [constructor](https://github.com/GPUOpen-Drivers/llpc/blob/dev/tool/vfx/vfxRenderSection.cpp#L24), where will [initialize the section info](https://github.com/GPUOpen-Drivers/llpc/blob/dev/tool/vfx/vfxRenderSection.cpp#L24).  This [uses](https://github.com/GPUOpen-Drivers/llpc/blob/dev/tool/vfx/vfxSection.h#L252) the variable [`Section::m_sectionInfo`](https://github.com/GPUOpen-Drivers/llpc/blob/dev/tool/vfx/vfxSection.h#L319), which is a static member.  This causes a segfault if the constructor for `Section::m_sectionInfo` has not been called yet.

Is there anything that guarantees the order in which they are initialized?

**Comment by s-perron** [comment]
This could be fixed by replacing `Section::m_sectionInfo` with a static function:

```
  static std::map<std::string, SectionInfo>& GetSectionInfo() {
    static std::map<std::string, SectionInfo> * sectionInfo = new std::map<std::string, SectionInfo>();
    return *sectionInfo;
  }
```

This will guarantee that the object is constructed before it is used.  However, the destructor will never be called, which can be a problem for tools that look for memory leaks.

**Comment by trenouf** [comment]
Is this anything to do with Qun's fix #767 ?

**Comment by s-perron** [comment]
Yes that looks like that will fix the problem.

**Comment by linqun** [comment]
Yes, it is my regression of PR#PR755. sorry for miss it. It should be fixed by PR#767.
In PR#767, I move static variable  renderSectionParserInit and vkSectionParserInit to functions , and call it explicitly when create document object, so that we can keep static variable is initialized in correct order.
